tensorflow_datasets/image/vis_dial.py

```diff
@@ -0,0 +1,136 @@
+  year={2017}
+}
+"""
+
+_TRAIN_DIALOGS = "https://dl.dropboxusercontent.com/s/ix8keeudqrd8hn8/visdial_1.0_train.zip"
+_TRAIN_IMAGES = "http://images.cocodataset.org/zips/train2014.zip"
+_VALIDATION_DIALOGS = "https://dl.dropboxusercontent.com/s/ibs3a0zhw74zisc/visdial_1.0_val.zip"
+_VALIDATION_IMAGES = "https://dl.dropboxusercontent.com/s/twmtutniktom7tu/VisualDialog_val2018.zip"
+_TEST_DIALOGS = "https://dl.dropboxusercontent.com/s/o7mucbre2zm7i5n/visdial_1.0_test.zip"
+_TEST_IMAGES = "https://dl.dropboxusercontent.com/s/mwlrg31hx0430mt/VisualDialog_test2018.zip"
+
+
+class VisualDialog(tfds.core.GeneratorBasedBuilder):
+  """Visual Dialog Dataset"""
+
+  VERSION = tfds.core.Version("1.0.0")
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description="A large set of Images and Dialogs",
+        features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(),
+            "dialog": tfds.features.SequenceDict({
+                "answer": tf.int8,
+                "question": tf.int8,
+            }),
+        }),
+        urls=["https://visualdialog.org/"],
+        citation=_CITATION
+    )
+
+
+  def _split_generators(self, dl_manager):
+
+    extracted_json = dl_manager.download_and_extract({
+        "Train_Dialogs": _TRAIN_DIALOGS,
+        "Validation_Dialogs": _VALIDATION_DIALOGS,
+        "Test_Dialogs": _TEST_DIALOGS,
+    })
+    extracted_images = dl_manager.download_and_extract({
+        "Train_Images": _TRAIN_IMAGES,
+        "Validation_Images": _VALIDATION_IMAGES,
+        "Test_Images": _TEST_IMAGES,
+    })
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=10,
+            gen_kwargs={
+                "image": extracted_images,
+                "dialog": extracted_json,
+            }),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            num_shards=10,
+            gen_kwargs={
+                "image": extracted_images,
+                "dialog": extracted_json,
+            }),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            num_shards=10,
+            gen_kwargs={
+                "image": extracted_images,
+                "dialog": extracted_json,
+            }),
+    ]
+
+  def _generate_examples(self, image, dialog):
+    """Generate examples as dicts.
+
+      Args:
+      image: `str`, directory containing the images folder
+      dialog: `str`, directory containing dialogs json
+
+      Yields:
+      Generator yielding the next samples
+    """
+
+    def find(pattern, path):
+      result = []
+      for root, dirs, files in tf.io.gfile.walk(path):
+        for name in files:
+          if fnmatch.fnmatch(name, pattern):
+            result.append(os.path.join(root, name))
+      return result
+
+    image_filedir = os.path.join(image["Test_Images"], "VisualDialog_test2018")
```

Comment:
This looks like you're always using "Test_images" - even when generate_examples is called for Train..

The tests would have caught it (as train and test splits would contain the same examples) - but it seems that you didn't create the test file ;-)